### DISCUSSION TITLE
BAC0 says "AttributeError: module 'BAC0' has no attribute 'connect'"

### DISCUSSION THREAD

**Discussion by vishalbacnet** [ChristianTremblay/BAC0#461] [Q&A]
Hi,

I have installed BAC0 library, i am jsut trying to run
        bacnet = BAC0.connect()
        bacnet.whois()

but getting error
bacpypes module missing, please install latest version using 
    $ "pip install bacpypes"

Discard this message if you are actually installing BAC0.
================================================================================
================================================================================
Import Error, refer to documentation or reinstall using 
    $ "pip install BAC0"
 No module named 'asyncore'

Discard this message if you are actually installing BAC0.

    bacnet = BAC0.connect()
             ^^^^^^^^^^^^
AttributeError: module 'BAC0' has no attribute 'connect'

Can someone please help hoe to resolve it.

**Comment by ChristianTremblay** [comment]
Python 3.12 removed asyncore from the library which means that you must now use bacpypes3 and the async version of BAC0.

you can test that by cloning the async branch. Pip install this branch

Then open the REPL by using

```
python -m asyncio
```

once in the REPL, you will need to adapt a little to the usage of await...

```
import BAC0
bacnet = BAC0.start()

bacnet.discover()

dev = await BAC0.device('2:5', 2005, bacnet) # create device from network 2, MAC 5, dev instance 2005

dev.points

val = await dev['objName'].value
```
